src/modalities/nn/weight_init/weight_init.py

```diff
@@ -0,0 +1,56 @@
+from typing import List
+
+import torch.nn as nn
+
+from modalities.nn.weight_init.weight_init_if import WeightInitializationIF
+
+
+class WeightInitializerWrapper(WeightInitializationIF):
+    def __init__(self, weight_initializers: List[WeightInitializationIF]):
+        self.weight_initializers = weight_initializers
+
+    def initialize_in_place(self, model: nn.Module):
+        for weight_initializer in self.weight_initializers:
+            weight_initializer.initialize_in_place(model)
+
+
+class ModulewiseNormalInitialization(WeightInitializationIF):
+    def __init__(self, mean: float, std: float):
+        """Initializes the weights of a model by sampling from a normal distribution.
+        NOTE: This class supports the initialization of nn.Linear and nn.Embedding layers.
+        For other layer types, the initialization must be subclassed and extended
+
+        Args:
+            mean (float): mean of the normal distribution
+            std (float): standard deviation of the normal distribution
+        """
+        self.mean = mean
+        self.std = std
+
+    def _init_weights_impl(self, module: nn.Module):
+        if isinstance(module, nn.Linear):
```

Comment:
Need to check how we can support SwiGLU here, with their different layers W, V, W2. 
https://arxiv.org/pdf/2002.05202

see here https://github.com/Modalities/modalities/blob/ec520f4dff20fdd3b68b79be7666b2fbb0e89f42/src/modalities/models/gpt2/gpt2_model.py#L395

Reply:
Also, the manual swiGLU implementation in main, has a linear layer called `c_proj` which is `V` in the original SwiGLU paper. So there is a naming conflict and we would apply `scaled init` to it, instead of `out_proj`, which corresponds to `W_2` in the paper and is the one that actually should receive `scaled_init`.

Reply:
We should probably rename the layers in our manual SwiGLU implementation in main, such that `out_proj = c_proj = W_2`, plus different names for `V` and `W`.

Reply:
see issue #170. We should fix this when we merge main into the branch of PR #145, since the SwiGLU reimplementation is only available in the latest main version.